worker/sshserver/worker_test.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sshserver_test
+
+import (
+	"go.uber.org/mock/gomock"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/controller"
+	"github.com/juju/juju/worker/sshserver"
+	"github.com/juju/juju/worker/sshserver/mocks"
+)
+
+type workerSuite struct {
+}
+
+var _ = gc.Suite(&workerSuite{})
+
+func newServerWrapperWorkerConfig(
+	l *mocks.MockLogger,
+	s *mocks.MockSystemStateGetter,
+	modifier func(*sshserver.ServerWrapperWorkerConfig),
+) *sshserver.ServerWrapperWorkerConfig {
+	cfg := &sshserver.ServerWrapperWorkerConfig{
+		NewServerWorker: func(sshserver.ServerWorkerConfig, bool) (*sshserver.ServerWorker, error) { return nil, nil },
+		Logger:          l,
+		StatePool:       s,
+		StateInfo: controller.StateServingInfo{
+			APIPort: 1234, // Fill in so it acts as non-zero value.
+		},
+	}
+
+	modifier(cfg)
+
+	return cfg
+}
+
+func (s *workerSuite) TestValidate(c *gc.C) {
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	mockLogger := mocks.NewMockLogger(ctrl)
+	mockStateGetter := mocks.NewMockSystemStateGetter(ctrl)
+
+	cfg := newServerWrapperWorkerConfig(mockLogger, mockStateGetter, func(cfg *sshserver.ServerWrapperWorkerConfig) {})
+	c.Assert(cfg.Validate(), gc.IsNil)
+
+	// Test no Logger.
+	cfg = newServerWrapperWorkerConfig(
+		mockLogger,
+		mockStateGetter,
+		func(cfg *sshserver.ServerWrapperWorkerConfig) {
+			cfg.Logger = nil
+		},
+	)
+	c.Assert(cfg.Validate(), gc.ErrorMatches, ".*is required.*")
+
+	// Test no StatePool.
+	cfg = newServerWrapperWorkerConfig(
+		mockLogger,
+		mockStateGetter,
+		func(cfg *sshserver.ServerWrapperWorkerConfig) {
+			cfg.StatePool = nil
+		},
+	)
+	c.Assert(cfg.Validate(), gc.ErrorMatches, ".*is required.*")
+
+	// Test no StateServingInfo.
+	cfg = newServerWrapperWorkerConfig(
+		mockLogger,
+		mockStateGetter,
+		func(cfg *sshserver.ServerWrapperWorkerConfig) {
+			cfg.StateInfo = controller.StateServingInfo{}
+		},
+	)
+	c.Assert(cfg.Validate(), gc.ErrorMatches, ".*is required.*")
+
+	// Test no NewServerWorker.
+	cfg = newServerWrapperWorkerConfig(
+		mockLogger,
+		mockStateGetter,
+		func(cfg *sshserver.ServerWrapperWorkerConfig) {
+			cfg.NewServerWorker = nil
+		},
+	)
+	c.Assert(cfg.Validate(), gc.ErrorMatches, ".*is required.*")
+}
```

Comment:
There are a few missing tests I'd like to see, you can see an example of them here:

https://github.com/juju/juju/blob/104d6edcbb9916994740d385822a9bd60c05bf61/worker/apicaller/manifold_test.go#L84-L96

https://github.com/juju/juju/blob/104d6edcbb9916994740d385822a9bd60c05bf61/worker/apicaller/manifold_test.go#L121-L129

Reply:
On the second one, is the .Stop() we want? My worker doesn't have a .Stop so I'm a bit confused. On the stub call I'm remvoing agent, we don't need it anymore. If that's what is meant here?